<launch>

  <arg name="lost" default="false"/>
  <arg name="robot" default="steel"/>
  <arg name="world" default="simple_office_with_people"/>
  <arg name="is_record_bag" default="false" />
  <arg name="bagfile" default="recorded_topics.bag" />
  <arg name="task_action_server_name" default="nursebot_actionserver" />
  <arg name="move_base_server_name" default="/move_base" />
  <arg name="head_ctrl_name" default="/head_controller/follow_joint_trajectory" />
  <arg name="torso_ctrl_name" default="/torso_controller/follow_joint_trajectory" />
  <arg name="gripper_ctrl_name" default="/gripper_controller/follow_joint_trajectory" />

  

  <include file="$(find nurse_bot)/launch/nursebot_gazebo.launch">
    <arg name="lost" value="$(arg lost)"/>
    <arg name="robot" value="$(arg robot)"/>
    <arg name="world" value="$(arg world)"/>
  </include>
  
  <param name="task_action_server_name" value="$(arg task_action_server_name)"/>
  <param name="move_base_server_name" value="$(arg move_base_server_name)"/>
  <param name="head_ctrl_name" value="$(arg head_ctrl_name)"/>
  <param name="torso_ctrl_name" value="$(arg torso_ctrl_name)"/>
  <param name="gripper_ctrl_name" value="$(arg gripper_ctrl_name)"/>


  <!-- <include file="$(find nurse_bot)/launch/nursebot_tasksubscriber.launch"/> -->
  <include file="$(find nurse_bot)/launch/nursebot_ui_node.launch"/>
  <include file="$(find nurse_bot)/launch/nursebot_action_server.launch"/>

   
   <group ns="nursebot">
    <node pkg="rosbag" type="record" name="rosbag_record" 
          args="record -O $(arg bagfile) -a"
          if="$(arg is_record_bag)"/>
   </group>

  <!-- <include file="$(find aruco_detect)/launch/aruco_detect.launch">
    <arg name="camera" value="/xtion/rgb" />
    <arg name="dictionary" value="16" />
    <arg name="fiducial_len" value=".045" />
    <arg name="image" value="image_raw" />
    <arg name="ignore_fiducials" value="0-581,583-1023" />
  </include> -->

  <!-- <node name="spawn_sdf" pkg="gazebo_ros" type="spawn_model" args="-sdf -file $(find tiago_gazebo)/models/aruco_cube/aruco_cube.sdf -model aruco_cube -x 2.1 -y -3 -z 0.68 -R 0.13 -P 1.57 -Y -2.98"  /> -->
  <!-- <node name="spawn_sdf" pkg="gazebo_ros" type="spawn_model" args="-sdf -file $(find tiago_gazebo)/models/aruco_cube/aruco_cube.sdf -model aruco_cube -x 2.57 -y -4.2 -z 0.46 -R 2.2 -P 1.57 -Y -0.97"  /> -->
  <node name="spawn_sdf" pkg="gazebo_ros" type="spawn_model" args="-sdf -file $(find tiago_gazebo)/models/aruco_cube/aruco_cube.sdf -model aruco_cube -x -3.98 -y 2.28 -z 0.48 -R -1.57 -P 0 -Y -1.59"  />
  <!-- -3.1 2.2 -->

  
</launch>